python/test/unit/language/test_frontend.py

```diff
@@ -0,0 +1,162 @@
+
+from filecheck.options import Options
+from filecheck.finput import FInput
+from filecheck.parser import Parser, pattern_for_opts
+from filecheck.matcher import Matcher
+
+import triton
+import triton.language as tl
+from triton.compiler import ASTSource, make_backend
+from triton.backends.compiler import GPUTarget
+from triton._C.libtriton import ir
+from triton.language.core import base_type, base_value
+
+# ===-----------------------------------------------------------------------===#
+# filecheck_test
+# ===-----------------------------------------------------------------------===#
+
+# Stub target for testing the frontend.
+stub_target = GPUTarget("cuda", 100, 32)
+stub_backend = make_backend(stub_target)
+
+llvm_bin_dir = os.path.join(os.path.dirname(sys.executable), "bin")
+filecheck_path = os.path.join(llvm_bin_dir, "FileCheck")
+
+
+def run_filecheck(name, module_str, check_template):
+    options = Options(match_filename=name)
+    fin = FInput(name, module_str)
+    ops = io.StringIO(check_template)
+    parser = Parser(options, ops, *pattern_for_opts(options))
+    matcher = Matcher(options, fin, parser)
+    matcher.stderr = io.StringIO()
+    if matcher.run() != 0:
+        raise ValueError(matcher.stderr.getvalue())
+
+
+def run_filecheck_test(kernel_fn):
+    assert isinstance(kernel_fn, triton.runtime.JITFunction)
+    check_template = inspect.getsource(kernel_fn.fn)
+    if check_template is None:
+        raise ValueError("kernel function must have a docstring with FileCheck template")
+    sigkeys = [x.name for x in kernel_fn.params]
+    sigvals = [f"arg{i}" for i in range(len(sigkeys))]
+    signature = {k: v for (k, v) in zip(sigkeys, sigvals)}
+    src = ASTSource(fn=kernel_fn, signature=signature)
+
+    context = ir.context()
+    ir.load_dialects(context)
+    stub_backend.load_dialects(context)
+
+    extra_options = src.parse_options()
+    options = stub_backend.parse_options(dict(**extra_options))
+    codegen_fns = stub_backend.get_codegen_implementation(options)
+    module_map = stub_backend.get_module_map()
+    mlir_module = src.make_ir(options, codegen_fns, module_map, context)
+
+    run_filecheck("placeholder", str(mlir_module), check_template)
+
+
+@triton.jit
+def anchor(v):
+    pass
+
+
+# Smoke test to make sure filecheck is working correctly.
+def test_filecheck_positive():
+
+    @triton.jit
+    def test_kernel():
+        # CHECK-LABEL: test_kernel
+        scalar = 42
+        # CHECK: %c42_i32 = arith.constant 42 : i32
+        # CHECK-NEXT: call @anchor{{.*}}(%c42_i32) : (i32) -> ()
+        anchor(scalar)
+
+    run_filecheck_test(test_kernel)
+
+
+def test_filecheck_negative():
+
+    @triton.jit
+    def test_kernel():
+        # CHECK-LABEL: test_kernel
+        scalar = 11
+        # CHECK: %c42_i32
+        anchor(scalar)
+
+    try:
+        run_filecheck_test(test_kernel)
+    except ValueError as e:
+        assert "Couldn't match \"%c42_i32\"" in str(e), "filecheck doesn't fail"
```

Comment:
This doesn't actually assert that an exception is raised, you need
```suggestion
    with pytest.raises(ValueError, match="Couldn't match \"%c42_i32\""):
        run_filecheck_test(test_kernel)
```

Reply:
nice catch, thanks!